### ISSUE TITLE
Completions on single colons

### ISSUE THREAD

**Issue by Veykril** [rust-lang/rust-analyzer#13597]
We seem to calculate completions for single colons of paths, we clearly shouldn't do that.

```rs
:$0 // shows completions
ResolvedThing:$0 // shows completions
UnresolvedThing:$0 // shows completions
```

**Comment by yue4u** [comment]
Not sure if I understand the issue correctly but `trigger_character` is None when the completion is triggered manually therefore following calculations happened. Related code:  https://github.com/rust-lang/rust-analyzer/blob/master/crates/rust-analyzer/src/handlers.rs#L815-L825

**Comment by Veykril** [comment]
Ye looks like we just need to handle the `None` case there as well
